src/lib.rs

```diff
@@ -534,27 +542,35 @@ impl<H, N, S, Id> HistoricalVotes<H, N, S, Id> {
 
 	/// Return the number of messages seen before prevoting.
 	/// None in case we didn't prevote yet.
-	pub fn prevote_idx(&self) -> Option<u64> {
-		self.prevote_idx
+	pub fn prevote_index(&self) -> Option<u32> {
+		self.prevote_index
 	}
 
 	/// Return the number of messages seen before precommiting.
 	/// None in case we didn't precommit yet.
-	pub fn precommit_idx(&self) -> Option<u64> {
-		self.precommit_idx
+	pub fn precommit_index(&self) -> Option<u32> {
+		self.precommit_index
 	}
 
 	/// Set the number of messages seen before prevoting.
-	pub fn set_prevoted_idx(&mut self) {
-		self.prevote_idx = Some(self.seen.len() as u64)
+	pub fn set_prevoted_index(&mut self) {
+		self.prevote_index = Some(self.seen.len() as u32)
 	}
 
 	/// Set the number of messages seen before precommiting.
-	pub fn set_precommited_idx(&mut self) {
-		self.precommit_idx = Some(self.seen.len() as u64)
+	pub fn set_precommited_index(&mut self) {
+		self.precommit_index = Some(self.seen.len() as u32)
 	}
 }
 
+pub trait AccountableSafety {
```

Comment:
Docs on public items.

Reply:
I need to see the substrate PR to understand what this trait is used for, seems it provides the same information we have on `HistoricalVotes`.